src/main/java/org/wildfly/security/ssl/SSLContextBuilder.java

```diff
@@ -315,8 +313,8 @@ public SSLContextBuilder setWrap(final boolean wrap) {
             final SSLContext sslContext = sslContextFactory.create();
             SSLSessionContext sessionContext = clientMode ? sslContext.getClientSessionContext() : sslContext.getServerSessionContext();
             if (sessionContext != null) {
-                sessionContext.setSessionCacheSize(sessionCacheSize);
-                sessionContext.setSessionTimeout(sessionTimeout);
+                if (sessionCacheSize > 0) sessionContext.setSessionCacheSize(sessionCacheSize);
```

Comment:
I think this needs to be >= 0 and ensure the default if never set is -1

Although we don't want the default to be '0' it is still a valid value that could be set.